packages/components/src/data-model/word-query/lexical-data/result/lexical-data-result.js

```diff
@@ -0,0 +1,16 @@
+export default class LexicalDataResult {
+  constructor (dataType) {
+    if (!dataType) {
+      throw new Error('LexicalDataResult cannot be created without a data type')
+    }
+
+    this.dataType = dataType
+    this.state = {
+      loading: false,
+      available: false,
+      failed: false
+    }
+    this.data = undefined
```

Comment:
`data` would usually contain the `HomonymGroup` but there might be something else in the future. I've decided to call it `data`, not `homonymGroup`, to keep things flexible.